roles/ansible_service_broker/tasks/upgrade.yml

```diff
@@ -6,8 +6,10 @@
     namespace: openshift-ansible-service-broker
   register: broker_configmap_raw
 
+- set_fact:
+    broker_configmap: '{{ (broker_configmap_raw.results.results.0.data | from_yaml)["broker-config"] | from_yaml }}'
+  when: broker_configmap_raw.results.results.0.data is defined
+
 - name: Migrate from etcd to CRDs
   import_tasks: migrate.yml
-  when: broker_configmap.dao.get('type') != 'crd'
-  vars:
-    broker_configmap: '{{ (broker_configmap_raw.results.results.0.data | from_yaml)["broker-config"] | from_yaml }}'
+  when: broker_configmap is defined and broker_configmap.dao.get('type') != 'crd'
```

Comment:
I broke this out to a set_fact. I could not get vars: and when: on the same task to play nice. I think it was trying to evaluating the vars before the when so short circuiting with something like this wasn't working:
`when: broker_configmap_raw.results.results.0.data is defined and broker_configmap.dao.get('type') != 'crd'`

If someone wants to take a stab at getting it to work like that feel free.